### DISCUSSION TITLE
Control Mode - Keyboard

### DISCUSSION THREAD

**Discussion by SJ824** [microsoft/AirSim#4876] [General]
When you first turn on airsim, you'll be controlling your car or drone via keyboard. However, I wanted to control it through code, so I wrote a simple executable code and connected it, but nothing happened in airsim when I ran the code. I thought it was probably because the control mode was set to keyboard. I would like to know how to change the control mode to API or automatic control mode. Or is there another workaround...!

**Comment by ghost** [comment]
When you use drone in Airsim, you won't be able to use keyboard for controls. You must use a joystick or an RC. Only with car you'll be able to control using your keyboard.

**Comment by EXPX3** [comment]
HI @SJ824 can you please explain your setup? are you using ROS? would like to know how your executable code to control drone communicates to the drone or car in airsim. Thanks in advance.
